### PR TITLE
journal: update to log 0.4

### PR DESCRIPTION
**Pull request by mathstuf**

### CODE REVIEW

File: src/journal.rs [hunk 1]
```diff
@@ -39,59 +39,50 @@ enum SyslogLevel {
...
-           &format!("CODE_FUNCTION={}", loc.module_path())]);
-}
+    let mut keys = vec![
+        format!("PRIORITY={}", lvl),
+        format!("MESSAGE={}", record.args()),
+        format!("TARGET={}", record.target()),
+    ];
 
-/// Record a log entry, with custom priority, location and target.
-pub fn log_target(level: usize, loc: &LogLocation, args: &fmt::Arguments, target: &str) {
```
Comment (mathstuf) [inline src/journal.rs]: With the removal of `LogLocation` for embedding directly into `Record`, these functions got unwieldy. If I should put them back, let me know.

File: README.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ fn main() {
    journal::send(["CODE_FILE=HI", "CODE_LINE=1213", "CODE_FUNCTION=LIES"]);
    journal::JournalLog::init().unwrap();
    warn!("HI");
-   sd_journal_log!(4, "HI {:?}", 2);
+   sd_journal_log!(log::Level::Warn, "HI {:?}", 2);
```
Comment (jmesmon) [inline README.md]: I'd really prefer to be able to pass the actual integer here still, so I suppose that means I'd prefer at least some of the logging functions be preserved.

**Comment by jmesmon** [comment]
I merged this, and added a commit on top to allow `sd_journal_log` to keep taking a `u32`. Also, it seemed that using `raw:`  as a macro selector didn't work (compiler complained about type ascription), so I changed `target:` to `@target` and added `@raw`.

**Comment by mathstuf** [comment]
For future reference, the change was made in 412f1141d2965402ddd8c63fac03ecba36cd07e1. One thing I note is that the syntax is `@target: <target>` and `@raw <raw>` (i.e., `@raw` doesn't have a colon). Which is preferred?

**Comment by jmesmon** [comment]
ah, I had meant to remove the `:` in both cases. I've changed it to that in 3015445edea144aa40760f6e4079f1a296d1b623

**Pull request merged**